modules/ingester/flush.go

```diff
@@ -165,49 +170,68 @@ func (i *Ingester) flushLoop(j int) {
 			return
 		}
 		op := o.(*flushOp)
+		op.attempts++
+
+		retry := false
 
-		var completeBlockID uuid.UUID
-		var err error
 		if op.kind == opKindComplete {
 			level.Debug(log.Logger).Log("msg", "completing block", "userid", op.userID)
-			instance, exists := i.getInstanceByID(op.userID)
-			if !exists {
-				// instance no longer exists? that's bad, log and continue
-				level.Error(log.Logger).Log("msg", "instance not found", "tenantID", op.userID)
+			instance, err := i.getOrCreateInstance(op.userID)
+			if err != nil {
+				handleFlushError(op, err)
 				continue
 			}
 
-			completeBlockID, err = instance.CompleteBlock(op.blockID)
-			if completeBlockID != uuid.Nil {
+			err = instance.CompleteBlock(op.blockID)
```

Comment:
I think the blockID for the `CompleteBlock` will not be the same as the completingBlock, we create a new uuid here - https://github.com/grafana/tempo/blob/fc4f477fb09507f54b39b9bdd75f594f0e9f2b42/tempodb/encoding/complete_block.go#L40

Reply:
This PR changes that method to keep the same uuid.